protocol/messenger_validate_requests_test.go

```diff
@@ -0,0 +1,45 @@
+package protocol
+
+import (
+	"testing"
+
+	"github.com/status-im/status-go/protocol/requests"
+
+	"github.com/stretchr/testify/suite"
+)
+
+func TestMessengerValidateRequestSuite(t *testing.T) {
+	suite.Run(t, new(MessengerValidateRequestSuite))
+}
+
+type MessengerValidateRequestSuite struct {
+	MessengerBaseTestSuite
+}
+
+func (s *MessengerValidateRequestSuite) TestSaveNewWakuNodeRequestValidate_Enrtree() {
+	alice := s.newMessenger()
+	r := requests.SaveNewWakuNode{NodeAddress: "enrtree://AIRVQ5DDA4FFWLRBCHJWUWOO6X6S4ZTZ5B667LQ6AJU6PEYDLRD5O@sandbox.waku.nodes.status.im"}
+	err := alice.SaveNewWakuNode(&r)
+	s.Require().NoError(err)
+}
+
+func (s *MessengerValidateRequestSuite) TestSaveNewWakuNodeRequestValidate_Multiaddr() {
+	alice := s.newMessenger()
+	r := requests.SaveNewWakuNode{NodeAddress: "/ip4/127.0.0.1/tcp/8080"}
+	err := alice.SaveNewWakuNode(&r)
+	s.Require().NoError(err)
+}
+
+func (s *MessengerValidateRequestSuite) TestSaveNewWakuNodeRequestValidate_MultiaddrFail() {
+	alice := s.newMessenger()
+	r := requests.SaveNewWakuNode{NodeAddress: "/0.0.0.0"}
+	err := alice.SaveNewWakuNode(&r)
+	s.Require().Error(err)
+}
+
+func (s *MessengerValidateRequestSuite) TestSaveNewWakuNodeRequestValidate_HttpFail() {
+	alice := s.newMessenger()
+	r := requests.SaveNewWakuNode{NodeAddress: "https://google.com"}
+	err := alice.SaveNewWakuNode(&r)
+	s.Require().Error(err)
+}
```

Comment:
Not a big deal, but 3/4 lines here are the same for each test.
This could be rewritten with test cases (you can find mane examples in out code searching for `testCases`):

```suggestion
func (s *MessengerValidateRequestSuite) TestSaveNewWakuNodeRequestValidate() {
    testCases := []struct {
		name                string
		nodeAddress   string
	}{
	    {
	        name: "ENRTree",
	        nodeAddress: "enrtree://AIRVQ5DDA4FFWLRBCHJWUWOO6X6S4ZTZ5B667LQ6AJU6PEYDLRD5O@sandbox.waku.nodes.status.im",
	    },
	    {
	        name: "Multiaddr",
	        nodeAddress: "/ip4/127.0.0.1/tcp/8080",
	    },
	    {
	        name: "MultiaddrFail",
	        nodeAddress: "/0.0.0.0",
	    },
	    {
	        name: "HttpFail",
	        nodeAddress: "https://google.com",
	    },
	}
	
	for _, tc := range testCases {
	    s.T().Run(tc.name, func() {
		    r := requests.SaveNewWakuNode{NodeAddress: tc.nodeAddress}
	        err := s.m.SaveNewWakuNode(&r)
	        s.Require().Error(err)
	    }
	}
}
```

Reply:
> Not a big deal, but 3/4 lines here are the same for each test. This could be rewritten with test cases (you can find mane examples in out code searching for `testCases`):

I think it's better for readability and independence to have small tests that only check one thing. But if you want, I can follow the current approach

https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29

Reply:
Well I don't have a strong opinion on this.
But I find it cleaner with test cases, because the test is dedicated to a single function. And splitting it into multiple functions just duplicates the code. If later you will change the arguments/returns of `SaveNewWakuNode`, you will have to modify 4 functions instead of 1 🤷‍♂️

Reply:
In my opinion, it's better to have different unit tests for different use cases of a single function. Easy to read and find the cause if it fails. Easy to write with fewer errors.

For this test, I don't think it matters. But for more complex functions, it's usually better to create a separate test rather than a single test to test all possible scenarios. 

Here I just remembered that some tests require Error, another NoError. I can add an "expectError" flag to the json